RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,974 @@
+ */
+
+#include "RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "DataFormats/Math/interface/deltaR.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+#include <memory>
+
+TSGForOIDNN::TSGForOIDNN(const edm::ParameterSet& iConfig)
+    : src_(consumes<reco::TrackCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      t_estimatorH_(esConsumes<Chi2MeasurementEstimatorBase, TrackingComponentsRecord>(
+          edm::ESInputTag("", iConfig.getParameter<std::string>("estimator")))),
+      t_magfieldH_(esConsumes<MagneticField, IdealMagneticFieldRecord>()),
+      t_propagatorAlongH_(esConsumes<Propagator, TrackingComponentsRecord>(
+          edm::ESInputTag("", iConfig.getParameter<std::string>("propagatorName")))),
+      t_propagatorOppositeH_(esConsumes<Propagator, TrackingComponentsRecord>(
+          edm::ESInputTag("", iConfig.getParameter<std::string>("propagatorName")))),
+      t_tmpTkGeometryH_(esConsumes<TrackerGeometry, TrackerDigiGeometryRecord>()),
+      t_geometryH_(esConsumes<GlobalTrackingGeometry, GlobalTrackingGeometryRecord>()),
+      t_navSchool_(esConsumes<NavigationSchool, NavigationSchoolRecord>(edm::ESInputTag("", "SimpleNavigationSchool"))),
+      t_SHPOpposite_(esConsumes<Propagator, TrackingComponentsRecord>(
+          edm::ESInputTag("", "hltESPSteppingHelixPropagatorOpposite"))),
+      maxSeeds_(iConfig.getParameter<uint32_t>("maxSeeds")),
+      maxHitSeeds_(iConfig.getParameter<uint32_t>("maxHitSeeds")),
+      maxHitlessSeeds_(iConfig.getParameter<uint32_t>("maxHitlessSeeds")),
+      numOfLayersToTry_(iConfig.getParameter<int32_t>("layersToTry")),
+      numOfHitsToTry_(iConfig.getParameter<int32_t>("hitsToTry")),
+      fixedErrorRescalingForHitless_(iConfig.getParameter<double>("fixedErrorRescaleFactorForHitless")),
+      minEtaForTEC_(iConfig.getParameter<double>("minEtaForTEC")),
+      maxEtaForTOB_(iConfig.getParameter<double>("maxEtaForTOB")),
+      updator_(new KFUpdator()),
+      measurementTrackerTag_(
+          consumes<MeasurementTrackerEvent>(iConfig.getParameter<edm::InputTag>("MeasurementTrackerEvent"))),
+      theCategory_(std::string("Muon|RecoMuon|TSGForOIDNN")),
+      maxHitlessSeedsIP_(iConfig.getParameter<uint32_t>("maxHitlessSeedsIP")),
+      maxHitlessSeedsMuS_(iConfig.getParameter<uint32_t>("maxHitlessSeedsMuS")),
+      maxHitDoubletSeeds_(iConfig.getParameter<uint32_t>("maxHitDoubletSeeds")),
+      getStrategyFromDNN_(iConfig.getParameter<bool>("getStrategyFromDNN")),
+      useRegressor_(iConfig.getParameter<bool>("useRegressor")),
+      etaSplitForDnn_(iConfig.getParameter<double>("etaSplitForDnn")),
+      dnnMetadataPath_(iConfig.getParameter<std::string>("dnnMetadataPath")) {
+  if (getStrategyFromDNN_) {
+    edm::FileInPath dnnMetadataPath(dnnMetadataPath_);
+    pt::read_json(dnnMetadataPath.fullPath(), metadata);
+    tensorflow::setLogging("2");
```

Comment:
why not the default `3`?

Reply:
We used the instructions from here: https://cms-ml.github.io/documentation/inference/tensorflow2.html.
At the bottom of the page it says that default log level is `1`. But I agree that we can set it to `3`.